src/Book.cpp

```diff
@@ -20,16 +20,37 @@ namespace LSMS
         mptr_parser = _ptr_parser;
     }
 
-    void Book::init(csv::Row *_ptr_row)
+    bool Book::init(csv::Row *_ptr_row)
     {
         if (!_ptr_row)
             throw csv::Error("Book: Row points to nullptr");
-        mptr_info = _ptr_row;
+        try
+        {
+            _ptr_row->get_headerptr()->getvalue("NAME"); //try if row is initialized with the right header
+        }
+        catch (const csv::Error &e)
+        { //must be the BID,DATE,POS
+            if (mptr_parser)
+            {
+                mptr_row = mptr_parser->find_first_of(_ptr_row->getvalue("BID"), "BID");
+                mptr_userinfo_style = _ptr_row;
+                return true;
+            }
+            else
+            {
+                mptr_row = nullptr;
+                mptr_userinfo_style = _ptr_row;
+                return false;
+            }
+        }
+
+        mptr_row = _ptr_row;
+        return true;
```

Comment:
The init function now checks if the Row* follows the style in Inventory.csv. If it doesn't the row is pointed to userinfo_style row and the proper row is searched for in the csv file